packages/cadl-ranch-api/src/request-validations.ts

```diff
@@ -34,6 +36,38 @@ export const validateBodyEquals = (request: RequestExt, expectedBody: unknown |
   }
 };
 
+export const validateXmlBodyEquals = (request: RequestExt, expectedBody: string): void => {
+  if (isBodyEmpty(request.rawBody)) {
```

Comment:
```suggestion
  if (request.rawBody === undefined || isBodyEmpty(request.rawBody)) {
```

that way you don't need the `!` below and it is accurate

Reply:
fixed